source/functions/test.txt

```diff
@@ -0,0 +1,313 @@
+      file in the ``functions`` directory for your function.
+
+      .. code-block:: sh
+
+         touch functions/hello.js
+
+
+      You also need to add configuration information for the Function to ``function/config.json``.
+
+      .. code-block:: json
+
+          {
+              "name": "hello",
+              "private": false,
+              "run_as_system": true
+          },
+
+      .. seealso::
+
+         For more information on creating a new Function,
+         refer to :ref:`Define a Function <define-a-function>`.
+
+   .. step:: Write Function code
+
+      To make your Function code easy to test, keep it modular
+      by separating its concerns into distinct components.
+      You must keep all logic for the Function in the file you defined
+      in the previous step. You cannot perform relative imports from other files
+      in your project in a Function file.
+      You can also :ref:`import dependencies using npm <external-dependencies>`.
+
+      You must export your function by assigning it to ``export``.
+
+      .. literalinclude:: /examples/generated/functions/hello.snippet.function.js
+         :language: js
+         :caption: hello.js
+         :emphasize-lines: 10
+
+   .. step:: Export Function for use in unit tests
+
+      To use your code to use in separate Node.js unit test files,
+      you must use separate CommonJS ``module.exports`` syntax.
+
+      This syntax is not compatible with the Functions runtime.
+      The Atlas Functions environment does not have the Node.js global ``module``.
+      To export modules to your unit tests while keeping the file compatible with
+      Functions, you must wrap the  the ``module.exports`` statement
+      with a check to see if the global ``module`` object exists.
+
+      .. literalinclude:: /examples/generated/functions/hello.snippet.function-with-local-export.js
+         :language: js
+         :caption: functions/hello.js
+         :emphasize-lines: 12-15
+
+   .. step:: Unit test exported Function code
+
+      Now you can write unit tests for the modules that you exported from the
+      Function file. Create a test file for the Function file.
+
+      Add your tests to a separate ``test`` directory in your project.
+
+      .. code-block:: sh
+
+         mkdir -p test/unit
+         touch test/unit/hello.test.js
+
+      Import the modules you exported in the previous step and add unit tests.
+
+      .. literalinclude:: /examples/generated/functions/hello.test.snippet.greeting-test.js
+         :language: js
+         :caption: test/unit/hello.test.js
+
+.. _functions_mock_services:
+
+Mock Services
+~~~~~~~~~~~~~
+
+To write unit tests for Functions that use the :ref:`global context object <function-context>`
+or one of the :ref:`other global modules <function-global-modules>`
+that Functions expose, you must create mocks of their behavior.
+
+In this example, the Function references an App Services Value via ``context.values.get()``.
```

Comment:
I would like to "App Services Values" here because many users might have no idea what that is. Since it's a prerequisite to understand this code (and learn about mocking for Functions) I think it's worth linking up here.